ethexe/common/src/tx_pool.rs

```diff
@@ -0,0 +1,132 @@
+// This file is part of Gear.
+//
+// Copyright (C) 2025 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! ethexe tx pool types
+
+use alloc::vec::Vec;
+use core::fmt;
+use gprimitives::{H160, H256};
+use parity_scale_codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+/// Ethexe transaction with a signature.
+#[derive(Clone, Encode, Decode, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SignedOffchainTransaction {
+    pub signature: Vec<u8>,
```

Comment:
what if we use here Signature type from ethexe_signer?

Reply:
Well, there's an invariant inside `Signature` and `RawSignature`, but it's not applied on a type level. So technically there are no issues with that, but from semantics perspective it looks wrong. Especially while having this: https://github.com/gear-tech/gear/blob/e482b4a27432b7f0a406783a4a34c199a3b71e85/ethexe/signer/src/signature.rs#L76

Reply:
discussed to be done in separate PR